### PR TITLE
Add resource name as an external id

### PR DESCRIPTION
**Pull request by clyang82**
fixed: https://github.com/openshift-online/maestro/issues/67

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -42,6 +42,7 @@ paths:
               schema:
                 $ref: '#/components/schemas/Error'
       parameters:
+      - $ref: '#/components/parameters/name'
```
Comment (skeeey) [inline openapi/openapi.yaml:45]: we may not need to introduce this parameter, it seems we can get resource with search parameter, e.g.
`/api/maestro/v1/resources?search=name = 'cluster3'`

Comment (clyang82) [reply openapi/openapi.yaml:45]: yes. search can be used for any field name. But it seems `/api/maestro/v1/resources?name=cluster3` is widely used.

File: pkg/services/types.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ type ListArguments struct {
 	Search   string
 	OrderBy  []string
 	Fields   []string
+	Name     string
```
Comment (morvencao) [inline pkg/services/types.go:18]: why does list argument contain `name`? can it just replace with search?

File: openapi/openapi.yaml (hunk 1 above)
Comment (clyang82) [reply openapi/openapi.yaml:45]: @markturansky How do you use in your side?

**Comment by clyang82** [comment]
remove query by name and  in favor of `ocm get /api/maestro/v1/resources --parameter search="name = 'nginx'"`

File: pkg/services/resource.go [hunk 3]
```diff
@@ -56,6 +56,11 @@ func (s *sqlResourceService) Get(ctx context.Context, id string) (*api.Resource,
 }
 
 func (s *sqlResourceService) Create(ctx context.Context, resource *api.Resource) (*api.Resource, *errors.ServiceError) {
+	if resource.Name != "" {
+		if err := ValidateResourceName(resource); err != nil {
+			return nil, errors.Validation("the name in the resource is invalid, %v", err)
+		}
+	}
```
Comment (qiujian16) [inline pkg/services/resource.go:63]: should we also check the name immutable when update?

Comment (clyang82) [reply pkg/services/resource.go:63]: Name string `gorm:"uniqueIndex"` there is error from database. we do not need to check here again for updating.

**Review by qiujian16 — APPROVED** [review]

File: pkg/services/resource.go (hunk 3 above)
Comment (clyang82) [reply pkg/services/resource.go:63]: the name is not exposed for patch. so the name will be ignored if it is changed when patch.

**Pull request merged**